mountpoint-s3/src/cli.rs

```diff
@@ -501,6 +477,36 @@ impl CliArgs {
     }
 }
 
+/// Generates a logging configuration based on the CLI arguments.
+///
+/// This includes random string generation which can change with each invocation,
+/// so once created the [LoggingConfig] should be cloned if another owned copy is required.
+fn logging_config(cli_args: &CliArgs) -> LoggingConfig {
+    let default_filter = if cli_args.no_log {
+        String::from("off")
+    } else {
+        let mut filter = if cli_args.debug {
+            String::from("debug")
+        } else {
+            String::from("warn")
+        };
+        let crt_verbosity = if cli_args.debug_crt { "debug" } else { "off" };
+        filter.push_str(&format!(",{}={}", AWSCRT_LOG_TARGET, crt_verbosity));
+        if cli_args.log_metrics {
+            filter.push_str(&format!(",{}=info", metrics::TARGET_NAME));
+        }
+        filter
+    };
+
+    let log_file = cli_args.log_directory.as_ref().map(|dir| prepare_log_file_name(dir));
+
+    LoggingConfig {
+        log_file,
+        log_to_stdout: cli_args.foreground,
+        default_filter,
+    }
+}
+
```

Comment:
I've moved this out of the `impl CliArgs` because it produces a different return value each time (due to randomness). I hope moving it out makes this clearer (along with RustDoc).

Reply:
Sounds good, that's much nicer. Updated.